src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -615,15 +615,12 @@ public void execute(LifecycleTransaction txn)
     public AllSSTableOpStatus performCleanup(final ColumnFamilyStore cfStore, int jobs) throws InterruptedException, ExecutionException
     {
         assert !cfStore.isIndex();
-
-        if (nodeHasPendingRangesForKeyspace(cfStore))
+        Keyspace keyspace = cfStore.keyspace;
+        if (!StorageService.instance.getTokenMetadata().getPendingRanges(keyspace.getName(), FBUtilities.getBroadcastAddressAndPort()).isEmpty())
```

Comment:
Thank you, the usage of `FBUtilities.getBroadcastAddress()` was wrong indeed - a consequence of sloppy propagation.

I think the idea of wrapping that not-very pleasant to-read predicate in an auxiliary method had some value but I am not going to oppose to removing it strongly.